Control/lib/control-core/RequestHandler.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @license
+ * Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+ * See http://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+ * All rights not expressly granted are reserved.
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+*/
+
+const {Log} = require('@aliceo2/web-ui');
+const log = new Log(`${process.env.npm_config_log_label ?? 'cog'}/controlrequests`);
+
+/**
+ * Handles AliECS create env requests
+ */
+class RequestHandler {
+
+  /**
+   * @param {object} ctrlService - Handle to Control service
+   */
+  constructor(ctrlService) {
+    this.ctrlService = ctrlService;
+    this.requestList = {};
+  }
+
+  /**
+   * Add AliECS request list to "cache", remove it from the "cache" once response comes from AliECS
+   * @param {object} req
+   * @param {object} res
+   */
+  async add(req, res) {
+    const index = Object.keys(this.requestList).length;
```

Comment:
I think here we are exposing ourselves to a potential bug. Lets assume we have 2 environments in creation. id is "0" and "1".
Then environment "0" is successfully created and removed from requests lists. We then have in our map { "1": {}}
Another user attempts to create an environment and this index will be assigned "1". Thus, the existing request will be overwritten. 
As you are using "indexes" I would recommend you use an array and then simply add to the end of it.

Reply:
I still think this is not good enough. For example
```
const requests = {}

let index = Object.keys(requests).pop() || 0;
// Even though you specify 0, JS will store it as a String Key
requests[index] = {key: 'some-request'}; // we add a request

index = Object.keys(requests).pop() || 0; // then we want to add another request
// this index will be a String '0' so it will once again override
```

Reply:
uhh, indeed, I've tested this now: `parseInt(Object.keys(obj).pop()) + 1 || 0;`